src/index.ts

```diff
@@ -1,48 +1,98 @@
+          this.checkHandlerName();
+          this.checkFunctionName();
+          this.checkHandlerNameMatchesFunction();
+
+          // If there were errors detected throw them all
+          if (this.errors.length !== 0) {
+               throw Error(this.errors.join('\r\n'));
+          }
+          
+          this.serverless.cli.log('Function check complete! No errors were found.');
+     }
+
+     // Handler name conventions
+     // Must be kebab-case (dash delimited)
+     // Must end in .handler
+     checkHandlerName() {
+          this.serverless.cli.log('Checking handler names');
 
           const functionNames = this.serverless.service.getAllFunctions();
           functionNames.forEach(fnName => {
-               // Get handler name
                const fn = this.serverless.service.getFunction(fnName) as Serverless.FunctionDefinitionHandler;
-               const handler = fn.handler.split('/').pop()?.slice(0, -8) as string;
+               // Get the full handler name (including path)
+               const handlerFullPath = fn.handler as string;
+               // Get the handler name with file extension but no path
+               const handlerName = handlerFullPath.split('/').pop() as string;
+               // Get the handler name with no path and no file extension
+               const handlerNameNoExtension = handlerName.replace('.handler', '') as string;
 
-               // Convert the function name and handler into values that can be compared (string[])
-               // Convert from camel case to an array of strings
-               const splitFn = fnName.split(/(?=[A-Z])/).map(s => s.toLowerCase());
-               // Convert from - to an array of strings
-               const splitHandler = handler.split('-');
+               // Check that the handler name is in kebab case
+               if (handlerNameNoExtension !== kebabCase(handlerNameNoExtension)) {
+                    this.errors.push(`Warning: Handler "${handlerName}" is not kebab case (dash delimited)`);
+               }
 
-               // Throw an error if the function does not match the handler name
-               if (JSON.stringify(splitFn) != JSON.stringify(splitHandler)) {
-                    // this.serverless.cli.log(`Warning: Function "${fnName}" does not match handler name "${handler}.handler"`);
-                    throw new Error(`Warning: Function "${fnName}" does not match handler name "${handler}.handler"`);
+               // Check that the handler ends in .handler
+               if (handlerNameNoExtension === handlerName) {
+                    this.errors.push(`Warning: Handler "${handlerName}" does not end in ".handler"`);
                }
           });
+     }
 
-          this.serverless.cli.log('Function check complete! No errors were found.');
+     // Function name conventions
+     // Must be camelCase
+     checkFunctionName() {
+          this.serverless.cli.log('Checking function names');
```

Comment:
I think that you can use `cli.debug` here to only output the more verbose logging when requested by the user.

Reply:
I can't find a debug option :disappointed: I might just remove these logs for now.